calculator_test.go

```diff
@@ -2,23 +2,416 @@ package calculator_test
+	}
+
+	got := calculator.Subtract(a, b...)
+	t.Logf("Random variadic test: Subtract(%v, %v), wants %v, got %v", a, b, want, got)
+	if got != want {
+		t.Errorf("want %v, got %v, while testing a random variadic case. The function call was: Subtract(%v, %v)", want, got, a, b)
+	}
+}
+
+func TestMultiply(t *testing.T) {
+	// Define test cases
+	testCases := []testCase{
+		{
+			description: "two positive numbers whos product is positive",
+			a:           2,
+			b:           20,
+			want:        40,
+		},
+		{
+			description: " a positive and negative number whos product is negative",
+			a:           7,
+			b:           -2,
+			want:        -14,
+		},
+		{
+			description: "a positive decimal and negative decimal whos product is a negative decimal",
+			a:           8.4,
+			b:           -2.5,
+			want:        -21,
+		},
+	}
+
+	t.Parallel()
+	for _, c := range testCases {
+		got := calculator.Multiply(c.a, c.b)
+		if got != c.want {
+			t.Errorf("want %v, got %v, while testing %s. The function call was: Multiply(%v, %v)", c.want, got, c.description, c.a, c.b)
+		}
+	}
+}
+
+// Randomly test the variadic call to Multiply()
+func TestMultiplyVariadicRandomly(t *testing.T) {
+	t.Parallel()
+	rand.Seed(time.Now().UnixNano())
+
+	// rand.Float64() returns a number in 0.0-1.0
+	// Use another randomly-generated int to vary the whole number.
+	a := rand.Float64() * float64(rand.Intn(500))
+	// b will be assigned multiple random float64 in a slice
+	b := randFloat64Slice(5)
+
+	// Get our own product to compare to Multiply()
+	var want float64 = a
+	for _, x := range b {
+		want = want * x
+	}
+
+	got := calculator.Multiply(a, b...)
+	t.Logf("Random variadic test: Multiply(%v, %v), wants %v, got %v", a, b, want, got)
+	if got != want {
+		t.Errorf("want %v, got %v, while testing a random variadic case. The function call was: Multiply(%v, %v)", want, got, a, b)
+	}
+}
+
+func TestDivide(t *testing.T) {
+	// Define test cases
+	testCases := []testCase{
+		{
+			description: "dividing by zero",
+			a:           2,
+			b:           0,
+			want:        123456789,
+			errExpected: true,
+		},
+		{
+			description: "two positive numbers whos quotient is positive",
+			a:           20,
+			b:           2,
+			want:        10,
+		},
+		{
+			description: " a positive and negative number whos quotient is negative",
+			a:           10,
+			b:           -2,
+			want:        -5,
+		},
+		{
+			description: "a positive decimal and negative decimal whos quotient is a negative decimal",
+			a:           8.4,
+			b:           -2.5,
+			want:        -3.3600000000000003,
+		},
+	}
+
+	t.Parallel()
+	for _, c := range testCases {
+		err, got := calculator.Divide(c.a, c.b)
+		if err != nil && c.errExpected == false {
+			t.Errorf("error received while testing %s. The function call was: Divide(%v, %v), and the error was: %v", c.description, c.a, c.b, err)
+		}
+
+		// Only fail on got != want if an error was not expected
+		if c.errExpected == false && got != c.want {
```

Comment:
Good! A shorter way to write this (as in the previous `if` statement) is: `!c.errExpected`.